playbooks/roles/ocp-config/tasks/main.yaml

```diff
@@ -56,6 +56,26 @@
     args:
       chdir: "{{ workdir }}"
 
+  - name: Generate Machine Config Master
+    template:
+      src: ../templates/99-master-machineconfig.yaml.j2
+      dest: "{{ workdir }}/openshift/99-master-machineconfig.yaml"
+    when:
+      - luks_compliant.enabled
```

Comment:
if you check if `luks_compliant is defined` you don't need to make it a required variable.

Reply:
@yussufsh As per our approach, we need to check if luks_compliant variable is enabled then only it will proceed with the MachineConfig creation task for both master and worker. Thus we have added the when condition to check it. Otherwise It will create the MachineConfig irrespective of the luks_compliant variable for all the cluster deployments
which is not required.
Also we are handling false positive test cases in powerVS terraform code for the same.
Please refer https://github.com/ocp-power-automation/ocp4-upi-powervs/blob/c2407b5a567c96aa7b99e10298de7a3680961892/modules/5_install/templates/install_vars.yaml

Reply:
Also as per your comment, in README we can keep luks_compliant.enabled for required column as no.

Reply:
`luks_compliant` should not be required. The condition should read as `luks_compliant is defined and luks_compliant.enabled`.
If we merge this code, anyone using the playbook will have errors saying luks_compliant is not defined. Hence from backporting point of view please add the check so that the playbook works even when I do not care about luks.

Reply:
@yussufsh As per your suggestion I've made changes in playbook. Please review it.